src/module.c

```diff
@@ -575,8 +575,10 @@ client *moduleAllocTempClient() {
     } else {
         c = createClient(NULL);
         c->flags |= CLIENT_MODULE;
-        c->user = NULL; /* Root user */
     }
+
+    c->user = user;
```

Comment:
are we sure we want to set the use in case it was taken from `moduleTempClients` ?

Reply:
all moduleTempClients reset to NULL at the end, so should be ok.